File: project/hyemin-onboarding/src/main/java/fastcampus/onboarding/form/controller/FormController.java
package fastcampus.onboarding.form.controller;

import fastcampus.onboarding.common.response.ApiResponse;
import fastcampus.onboarding.form.dto.request.FormCreateRequestDto;
import fastcampus.onboarding.form.dto.request.FormUpdateRequestDto;
import fastcampus.onboarding.form.service.FormService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;


@RestController
@RequiredArgsConstructor
@Slf4j
public class FormController {
    private final FormService formService;

    @PostMapping("/form")
    public ResponseEntity<ApiResponse<Void>> createForm(@Valid @RequestBody FormCreateRequestDto formCreateRequestDto) {
        try {

            formService.createForm(formCreateRequestDto);
            return ResponseEntity.status(HttpStatus.CREATED)
                    .body(ApiResponse.success(HttpStatus.CREATED, "설문조사 생성을 성공하였습니다."));
        } catch (Exception e) {
            return ResponseEntity
                    .status(HttpStatus.INTERNAL_SERVER_ERROR)
                    .body(ApiResponse.error(HttpStatus.INTERNAL_SERVER_ERROR, "서버 내부 오류"));
        }
    }

    @PutMapping("/form/{formSeq}")
    public ResponseEntity<ApiResponse<Void>> updateForm(@PathVariable Long formSeq, @RequestBody FormUpdateRequestDto formUpdateRequestDto) {
        try {
            formService.updateForm(formSeq, formUpdateRequestDto);
            return ResponseEntity.status(HttpStatus.OK)
                    .body(ApiResponse.success(HttpStatus.OK, "설문조사를 수정하였습니다."));
        } catch (Exception e) {
            return ResponseEntity
                    .status(HttpStatus.INTERNAL_SERVER_ERROR)
                    .body(ApiResponse.error(HttpStatus.INTERNAL_SERVER_ERROR, "서버 내부 오류"));
        }
    }
}

File: project/sanghyuk-onboarding/src/main/kotlin/fc/innercircle/sanghyukonboarding/form/interfaces/rest/FormController.kt
package fc.innercircle.sanghyukonboarding.form.interfaces.rest

import fc.innercircle.sanghyukonboarding.form.application.CreateFormFacade
import fc.innercircle.sanghyukonboarding.form.application.EditFormFacade
import fc.innercircle.sanghyukonboarding.form.domain.model.Form
import fc.innercircle.sanghyukonboarding.form.interfaces.rest.port.dto.request.FormRequest
import fc.innercircle.sanghyukonboarding.form.interfaces.rest.port.dto.response.FormResponse
import fc.innercircle.sanghyukonboarding.form.service.port.FormQueryRepository
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.servlet.support.ServletUriComponentsBuilder.fromCurrentRequest
import java.net.URI

@RequestMapping(path = ["/api/v1/forms"])
@RestController
class FormController(
    private val createFormFacade: CreateFormFacade,
    private val editFormFacade: EditFormFacade,
    private val formQueryRepository: FormQueryRepository
) {

    @PostMapping(consumes = ["application/json"], produces = ["application/json"])
    fun createForm(@RequestBody command: FormRequest): ResponseEntity<Unit> {
        val formId: String = createFormFacade.create(command)
        val location: URI = fromCurrentRequest()
            .path("/{id}")
            .buildAndExpand(formId)
            .toUri()
        return ResponseEntity.created(location).build()
    }

    @GetMapping("/{formId}", produces = ["application/json"])
    fun getById(@PathVariable formId: String): ResponseEntity<FormResponse> {
        val form: Form = formQueryRepository.getById(formId)
        val response: FormResponse = FormResponse.from(form)
        return ResponseEntity.ok(response)
    }

    @PutMapping("/{formId}", consumes = ["application/json"], produces = ["application/json"])
    fun editForm(
        @PathVariable formId: String,
        @RequestBody command: FormRequest
    ): ResponseEntity<Unit> {
        editFormFacade.edit(formId, command)
        return ResponseEntity.noContent().build()
    }
}
